torch/csrc/distributed/autograd/context/dist_autograd_container.cpp

```diff
@@ -33,35 +41,69 @@ DistAutogradContainer& DistAutogradContainer::getInstance() {
   return container;
 }
 
-const DistAutogradContext& DistAutogradContainer::newContext() {
-  if (!initialized_) {
-    throw std::runtime_error(
-        "Need to initialize distributed autograd using "
-        "torch.distributed.autograd.init()");
+int64_t DistAutogradContainer::newAutogradMessageId() {
+  checkInitialized();
+
+  // Check for int64_t max overflow.
+  TORCH_INTERNAL_ASSERT(
+      next_autograd_message_id_ < std::numeric_limits<int64_t>::max());
+  // Check for overflow into workerId_ section.
+  TORCH_INTERNAL_ASSERT(next_autograd_message_id_ < max_id_);
+  return next_autograd_message_id_++;
+}
+
+DistAutogradContext& DistAutogradContainer::createContext(int64_t context_id) {
+  checkInitialized();
+
+  std::lock_guard<std::mutex> guard(autograd_context_lock_);
+  auto it = autograd_context_.find(context_id);
+  if (it != autograd_context_.end()) {
+    return it->second;
   }
 
+  auto& context = autograd_context_
```

Comment:
`newContext` checks the value of `next_context_id_ `, but this method does not. Is it possible that someone pass in a `std::numeric_limits<int64_t>::max()` to this method and breaks the invariant?